src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -4703,6 +4705,8 @@ def update_image_version(cmd, resource_group_name, gallery_name, gallery_image_n
         image_version.publishing_profile.replica_count = replica_count
     if image_version.storage_profile.source is not None:
         image_version.storage_profile.os_disk_image = image_version.storage_profile.data_disk_images = None
+    # target extended locations are not contained if the update command is called without target edge zones
+    image_version.publishing_profile.target_extended_locations = target_edge_zones
```

Comment:
When using the `az sig image-version update` command, if the customer does not pass in the `--target-edge-zones` parameter, the existing target edge zones will be cleared. This behavior may be a bug for users, the main reasons are as follows:
1. When users use the `az sig image-version update` command to update other properties, they may not consider that the target edge zones will be affected. It is an unexpected behavior for them to clear target edge zones.
2. This approach is inconsistent with the common design of other CLI commands. We usually recommend that the user specify a special value to indicate emptying the existing propertites. If the user does not pass in the relevant parameters, this property will not be updated. Such as the `--nsg-rule NONE` for `az vm create` indicates that users want to clear the nsg rule.

So this design which looks different from other commands looks more like a bug. So I suggest keeping the design consistent with other commands to let the user explicitly enter `--target-edge-zones NONE` to indicate clearing the existing target edge zones.